src/test/java/org/broadinstitute/hellbender/utils/config/ConfigUnitTest.java

```diff
@@ -0,0 +1,277 @@
+                    new ArrayList<>(Arrays.asList(new String[] {"main","END"})),
+                },
+        };
+    }
+
+    @DataProvider
+    Object[][] createArgsAndConfigFileOptionsBadInput() {
+        return new Object[][] {
+                {
+                        new ArrayList<>(Arrays.asList(new String[] {"main", "testArg", "--config"})),
+                        "--config",
+                },
+                {
+                        new ArrayList<>(Arrays.asList(new String[] {"main","--config"})),
+                        "--config",
+                },
+                {
+                        new ArrayList<>(Arrays.asList(new String[] {"--config"})),
+                        "--config",
+                },
+        };
+    }
+
+    // ================================================================================
+    // Tests:
+    // ================================================================================
+
+    @Test(dataProvider = "createArgsAndConfigFileOptionsBadInput",
+            expectedExceptions = UserException.BadInput.class)
+    void testGetConfigFilenameFromArgsBadInput( final ArrayList<String> args,
+                                        final String configFileOption) {
+
+        ConfigUtils.getConfigFilenameFromArgs(args, configFileOption);
+    }
+
+    @Test(dataProvider= "createArgsAndConfigFileOptions")
+    void testGetConfigFilenameFromArgs( final ArrayList<String> args,
+                                        final String configFileOption,
+                                        final String expectedFilename,
+                                        final ArrayList<String> expectedRemainingArgs) {
+
+        String outFileName = ConfigUtils.getConfigFilenameFromArgs(args, configFileOption);
+
+        Assert.assertEquals(expectedFilename, outFileName);
+        Assert.assertEquals(expectedRemainingArgs, args);
+    }
+
+    @Test
+    void testInitializeConfiguration() throws IOException {
+        String inputPropertiesFile = testResourceDir + "AdditionalTestOverrides.properties";
+
+        BasicTestConfig basicTestConfig = ConfigUtils.initializeConfigurationAsProperties(inputPropertiesFile, BasicTestConfig.class);
+
+        // List properties for inspection:
+        listAndStoreConfigToStdOut(basicTestConfig);
+
+        // Check for our values:
+        Assert.assertEquals(basicTestConfig.booleanDefFalse(), false);
+        Assert.assertEquals(basicTestConfig.booleanDefTrue(), true);
+        Assert.assertEquals(basicTestConfig.intDef207(), 999);
+        Assert.assertEquals(basicTestConfig.listOfStringTest(), new ArrayList<>(Arrays.asList(new String[] {"string1", "string2", "string3", "string4"})));
+
+    }
+
+    @Test
+    void testSystemConfiguration() {
+        // Test with our basic test class:
+        SystemTestConfig testConfig = ConfigFactory.create(SystemTestConfig.class);
+
+        ConfigUtils.injectSystemPropertiesFromConfig(testConfig);
```

Comment:
You probably want to assert that these were unset before the injection. Otherwise there might be confusing issues if someone reuses the same property names in another test.